### PR TITLE
Lint warnings

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28550

## Description
added support for warning messages report .
image

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/lint/helpers.py [hunk 1]
```diff
@@ -407,3 +407,20 @@ def pylint_plugin(dest: Path):
...
+        """
+    output_lst = output.split('\n')
+    warnings_list = []
+    error_list = []
+    for msg in output_lst:
+        if msg.startswith('W') or 'W90' in msg:
+            warnings_list.append(msg)
+        elif msg.startswith('E') or 'E90' in msg:
+            error_list.append(msg)
+    return error_list, warnings_list
```
Comment (GalRabin) [inline demisto_sdk/commands/lint/helpers.py:426]:
```suggestion
            error_list.append(msg)
            
    return error_list, warnings_list
```

File: demisto_sdk/commands/lint/lint_manager.py [hunk 2]
```diff
@@ -260,6 +260,16 @@ def run_dev_packages(self, parallel: int, no_flake8: bool, no_xsoar_linter: bool
...
+            "warning_packs_flake8": [],
+            "warning_packs_XSOAR_linter": [],
+            "warning_packs_bandit": [],
+            "warning_packs_mypy": [],
+            "warning_packs_vulture": [],
+            "warning_packs_pylint": [],
+            "warning_packs_pytest": [],
+            "warning_packs_pwsh_analyze": [],
+            "warning_packs_pwsh_test": [],
+            "warning_packs_image": [],
```
Comment (GalRabin) [inline demisto_sdk/commands/lint/lint_manager.py:272]: There is not used for warning in all other lint checks.

File: demisto_sdk/commands/lint/lint_manager.py [hunk 3]
```diff
@@ -276,6 +286,7 @@ def run_dev_packages(self, parallel: int, no_flake8: bool, no_xsoar_linter: bool
 
         with concurrent.futures.ThreadPoolExecutor(max_workers=parallel) as executor:
             return_exit_code: int = 0
+            return_warning_code: int = 0
```
Comment (GalRabin) [inline demisto_sdk/commands/lint/lint_manager.py:289]: Can't we just print if there are warnings?  All other changes related to the report looks good, just need to verify some edge case, with warnings/without warnings/ warnings and errors/Empty (Which I believe you did).

**Comment by orhovy** [comment]
for -a runs of lint the warning messages are not set to not be printed in the report as there will be a lot of warnings all over the repo. @bakatzir

File: demisto_sdk/commands/lint/lint_manager.py [hunk 4]
```diff
@@ -565,16 +612,22 @@ def report_summary(self, lint_status: dict):
                                                  subsequent_indent=' ' * len(fail_pack_prefix))
         # intersection of all failed packages
         failed: Set[str] = set()
-        for packs in lint_status.values():
-            failed = failed.union(packs)
+        warnings: Set[str] = set()
+        for key in lint_status:
+            if key.startswith('fail'):
+                failed = failed.union(lint_status[key])
+            if key.startswith('warning'):
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/lint_manager.py:619]: If it failed, it is ok that this apck has warnings, but we want to report it as a failure. right?
```suggestion
           elif key.startswith('warning'):
```

File: demisto_sdk/commands/lint/lint_manager.py [hunk 5]
```diff
@@ -565,16 +612,22 @@ def report_summary(self, lint_status: dict):
...
         sentence = " Summary "
         print(f"\n{Colors.Fg.cyan}{'#' * len(sentence)}")
         print(f"{sentence}")
         print(f"{'#' * len(sentence)}{Colors.reset}")
-        print(f"Packages: {len(self._pkgs)}")
-        print(f"Packages PASS: {Colors.Fg.green}{len(self._pkgs) - len(failed)}{Colors.reset}")
+        print(f"Packages: {len(pkg)}")
+        print(f"Packages PASS: {Colors.Fg.green}{len(pkg) - len(failed)}{Colors.reset}")
         print(f"Packages FAIL: {Colors.Fg.red}{len(failed)}{Colors.reset}")
+        print(f"Packages WARNING: {Colors.Fg.yellow}{len(warnings)}{Colors.reset}")
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/lint_manager.py:629]: Need to make sure this is true. as pkg can pass with warning, and pkg can fail and also have warnings
```suggestion
        print(f"Packages PASS: {Colors.Fg.green}{len(pkg) - len(failed) - len(warnings)}{Colors.reset}")
        print(f"Packages PASS WITH WARNINGS: {Colors.Fg.yellow}{len(warnings)}{Colors.reset}")
        print(f"Packages FAIL: {Colors.Fg.red}{len(failed)}{Colors.reset}")

```

**Review by bakatzir — COMMENTED** [review]
Very nice. I might have missed the usage of the non xsoar_linter warnings, can you elaborate or attach an example?

File: demisto_sdk/commands/lint/lint_manager.py (hunk 4 above)
Comment (orhovy) [reply demisto_sdk/commands/lint/lint_manager.py:619]: yes , but we still want to show its warnings too, if there are

File: demisto_sdk/commands/lint/lint_manager.py (hunk 5 above)
Comment (orhovy) [reply demisto_sdk/commands/lint/lint_manager.py:629]: I was going for the option that no matter what if there are warnings than they will be printed and will be added to the summary report. However they do not fail lint in any way. 
Also , Fail and warnings can be shown together. For example, xsoar linter runs and there are failures as print exists in the code, but I would still want to show the warnings which are that there are still `demisto.results` usage and no main.

File: demisto_sdk/commands/lint/lint_manager.py (hunk 4 above)
Comment (bakatzir) [reply demisto_sdk/commands/lint/lint_manager.py:619]: ok, let's make sure we male it clear.
e.g - passing packs with warnings, and failed packs with warnings.

File: demisto_sdk/commands/lint/helpers.py [hunk 6]
```diff
@@ -413,6 +413,12 @@ def split_warnings_errors(output: str):
     """
         Function which splits the given string into warning messages and error using W or E in the beginning of string
         For error messages that do not start with E , they will be returned as other.
+        The output of a certain pack can both include:
+            - Fail msgs
+            - Fail msgs and warnings msgs
+            - Passed msgs
+            - Passed msgs and warninsg msgs
```
Comment (bakatzir) [inline demisto_sdk/commands/lint/helpers.py:420]:
```suggestion
            - Passed msgs and warnings msgs
```

File: demisto_sdk/commands/lint/lint_manager.py (hunk 5 above)
Comment (bakatzir) [reply demisto_sdk/commands/lint/lint_manager.py:634]: So let's just add here
```
print(f"Packages WARNING(can either PASS or FAIL): {Colors.Fg.yellow}{len(warnings)}{Colors.reset}")
```

**Review by bakatzir — APPROVED** [review]
Let's ask a few team members regarding the clearer "packs WARNINGS" msg and find a GCD.
Aside from that, looks GREAT.

**Comment by bakatzir** [comment]
coveralls jitters, cov is up from this PR as shown from prev runs.

**Pull request merged**